### PR TITLE
[Graphbolt] Add overview in doc

### PR DESCRIPTION
**Pull request by peizhou001**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: docs/source/stochastic_training/index.rst [hunk 1]
```diff
@@ -7,6 +7,37 @@ scalability. It is built on top of DGL and PyTorch.
...
+^^^^^^^
+
+.. image:: ../../_static/graphbolt_overview.png
+  :width: 700
+  :alt: Graphbolt Overview
+
+GraphBolt seamlessly integrates with the PyTorch `datapiple <https://pytorch.org/data/beta/torchdata.datapipes.iter.html>`_, where each stage's output is just the input of its successor, streamlining data loading and preprocessing for GNN training, validation and testing.
+By default, GraphBolt provides a collection of built-in datasets and exceptionally efficient implementations of datapipes for common scenarios, which can be summarized as follows:
+
+1. **Mini-Batch Sampler:** Randomly selects a subset (nodes, edges, graphs) from the entire training set to calculate embeddings.
```
Comment (Rhett-Ying) [inline docs/source/stochastic_training/index.rst:20]: One may be confusing is such class is named as `ItemSampler` in GB.

File: docs/source/stochastic_training/index.rst [hunk 2]
```diff
@@ -7,6 +7,37 @@ scalability. It is built on top of DGL and PyTorch.
...
+
+2. **Negative Sampler:** In link prediction tasks, samples non-existing edges as negative examples to train the GNN in distinguishing between positive and negative edges.
+
+3. **Subgraph Sampler:** Utilizes the output from mini-batch sampling to generate neighbor nodes up to a specified depth or distance.
```
Comment (frozenbugs) [inline docs/source/stochastic_training/index.rst:24]: Does it have to be neighbor nodes? What about induced subgraph? Can you rephrase it and focus on just about *generating subgraph*.

File: docs/source/stochastic_training/index.rst [hunk 3]
```diff
@@ -7,6 +7,37 @@ scalability. It is built on top of DGL and PyTorch.
 This tutorial introduces how to enable stochastic training of GNNs with
 GraphBolt.
 
+Overview
+^^^^^^^
+
+.. image:: ../../_static/graphbolt_overview.png
```
Comment (frozenbugs) [inline docs/source/stochastic_training/index.rst:13]: The image is missing

**Comment by peizhou001** [comment]
image

**Review by frozenbugs — APPROVED** [review]
